test/integration/host_test.rb

```diff
@@ -533,6 +475,29 @@ def test_show_action_buttons
       host.reload
       assert_equal env1.name, host.environment.name
     end
+
+    test 'enables override of hostgroup defaults' do
+      original_environment = FactoryGirl.create(:environment)
+      original_hostgroup = FactoryGirl.create(:hostgroup,
+                                              :environment => original_environment)
+
+      # Clear host environment, then inherit it from hostgroup
+      @host.environment = nil
+      @host.hostgroup = original_hostgroup
+      @host.send(:assign_hostgroup_attributes, ['environment_id'])
+      @host.save
+
+      overridden_hostgroup = FactoryGirl.
+        create(:hostgroup, :environment => FactoryGirl.create(:environment))
+
+      visit edit_host_path(@host)
```

Comment:
Indeed, I think that matches my comment at https://github.com/theforeman/foreman/pull/3001#issuecomment-167083945.

Perhaps this PR should keep the "new" test then, but without expecting the original environment to change rather than moving it to "edit" - or even better, keep both.